src/SonarScanner.MSBuild.Common/MsBuildPathSettings.cs

```diff
@@ -48,7 +76,12 @@ public IEnumerable<string> GetImportBeforePaths()
 
         public IEnumerable<string> GetGlobalTargetsPaths()
         {
-            var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+            var programFiles = this.getEnvironmentSpecialFolderPath(Environment.SpecialFolder.ProgramFiles, false);
+
+            if (string.IsNullOrEmpty(programFiles))
+            {
+                throw new IOException("The program files folder doesn't exist and it was not possible to create it.");
```

Comment:
It doesn't matter much, but why is the message saying `it was not possible to create it`? Perhaps cut that part of both messages?